web/src/app/modules/views/main/editors/modules/graphical-editor/elements/graphical-element-base.ts

```diff
@@ -18,6 +18,7 @@ export abstract class GraphicalElementBase<T extends IContainer> {
     }
 
     public get isValid(): boolean {
-        return this.validationService.isValid(this.element);
+        return this.validationService.isValid(this.element) &&
+            this.validationService.currentInvalidElements.find((element: IContainer) => element === this.element) === undefined;

```

Comment:
probably there is good reason, but I don't get why is there an isValid method AND the currentInvalidElements collection? This seems redundant.

Reply:
Yes, there is: we perform the isValid check not on each element seperately for performance reasons. We rather check, e.g., the model and its contents, and not all its children seperately, because for validating the model, we have to check its children anyways. I moved the check, however.